deicode/tests/test_optspace.py

```diff
@@ -18,8 +21,22 @@
 
 class TestOptspace(unittest.TestCase):
     def setUp(self):
+        self.obs = model_data()
         pass
 
+    def test_rank_estimation(self):
+        """Test rank estimation is accurate."""
+        # calc. rclr
+        obs = rclr(self.obs)
+        obs[~np.isfinite(obs)] = 0.0
+        # eps (calculated in OptSpace)
+        m, n = obs.shape
+        mask = (np.abs(obs) > 0).astype(np.int)
+        total_nonzeros = np.count_nonzero(mask)
+        eps = total_nonzeros / np.sqrt(m * n)
+        # estimate rank
+        self.assertEqual(2, rank_estimate(obs, eps))
```

Comment:
I don't understand this test ...

If I were to do this, I would do something like
```python
N = 100
D = 50
k = 3
U = np.random.randn(0, 1, size=(N, k))
V = np.random.randn(0, 1, size=(k,D))
Y = U @ V
# randomly mask Y
mask = np.random.random(size=(N, D))
Y[mask] = 0
self.assertEqual(k, rank_estimate(Y, eps))
```

Reply:
Agree, I changed the test accordingly.